out_writeapi.go

```diff
@@ -160,6 +162,20 @@ func checkResponses(curr_ctx context.Context, currQueuePointer *[]*managedwriter
 	return 0
 }
 
+func sendRequest(ctx context.Context, data [][]byte, config **outputConfig) error {
+	(*config).mutex.Lock()
+	defer (*config).mutex.Unlock()
```

Comment:
It would make sense to move the mutex lock inside the if statement.

Reply:
Is there a particular reason why? As in, why would did we put the mutex lock outside of the for loop in checkResponses but inside the if statement in sendRequest?